crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2288,16 +2288,19 @@ impl Client {
 
     /// Repeatedly synchronize the client state with the server.
     ///
-    /// This method will never return, if cancellation is needed the method
-    /// should be wrapped in a cancelable task or the
-    /// [`Client::sync_with_callback`] method can be used.
+    /// This method will only return on error, if cancellation is needed
```

Comment:
I wonder if we should describe a bit more around the errors here. I think it's a bit non-obvious that usually requests are retried, so sporadic network failures will still be handled by the `Client` and the user doesn't need to care about those.